pkg/gce-pd-csi-driver/controller.go

```diff
@@ -152,7 +152,7 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
 	content := req.GetVolumeContentSource()
 	if content != nil {
 		if content.GetSnapshot() != nil {
-			snapshotId = content.GetSnapshot().GetId()
+			snapshotId = content.GetSnapshot().GetSnapshotId()
 		}
 	}
```

Comment:
Modify `ControllerServiceCapability` to return new `PUBLISH_READONLY` capability.

See https://github.com/container-storage-interface/spec/pull/310/files for details